docs/guides/ingesters-rolling-updates.md

```diff
@@ -0,0 +1,91 @@
+
+- [Blocks storage](#blocks-storage)
+- [Chunks storage with WAL enabled](#chunks-storage-with-wal-enabled)
+- [Chunks storage with WAL disabled](#chunks-storage-with-wal-disabled-hand-over)
+
+## Blocks storage
+
+The Cortex [blocks storage](../blocks-storage/_index.md) requires ingesters to run with a persistent disk where the TSDB WAL and blocks are stored (eg. a StatefulSet when deployed on Kubernetes).
+
+During a rolling update, the leaving ingester closes the open TSDBs, synchronize the data to disk (`fsync`) and releases the disk resources.
+The new ingester, which is expected to reuse the same disk of the leaving one, will replay the TSDB WAL on startup in order to load back in memory the time series that have not been compacted into a block yet.
+
+_The blocks storage doesn't support the series [hand-over](#chunks-storage-with-wal-disabled-hand-over)._
+
+## Chunks storage
+
+The Cortex chunks storage optionally supports a write-ahead log (WAL).
+The rolling update procedure for a Cortex cluster running the chunks storage depends whether the WAL is enabled or not.
+
+### Chunks storage with WAL enabled
+
+Similarly to the blocks storage, when Cortex is running the chunks storage with WAL enabled, it requires ingesters to run with a persistent disk where the WAL is stored (eg. a StatefulSet when deployed on Kubernetes).
+
+During a rolling update, the leaving ingester closes the WAL, synchronize the data to disk (`fsync`) and releases the disk resources.
+The new ingester, which is expected to reuse the same disk of the leaving one, will replay the WAL on startup in order to load back in memory the time series that have not been flushed to the storage yet.
+
+_For more information about the WAL, please refer to [Ingesters with WAL](../production/ingesters-with-wal.md)._
+
+### Chunks storage with WAL disabled (hand-over)
+
+When Cortex is running the chunks storage with WAL disabled, Cortex supports on-the-fly series hand-over between a leaving ingester and a joining one.
+
+The hand-over is based on the ingesters state stored in the ring. Each ingester could be in one of the following **states**:
+
+- `PENDING`
+- `JOINING`
+- `ACTIVE`
+- `LEAVING`
+
+On startup, an ingester goes into the **`PENDING`** state.
+In this state, the ingester is waiting for a hand-over from another ingester that is `LEAVING`.
+If no hand-over occurs within the configured timeout period ("auto-join timeout", configurable via `-ingester.join-after` option), the ingester will join the ring with a new set of random tokens (eg. during a scale up) and will switch its state to `ACTIVE`.
+
+A running ingester, in the **`ACTIVE`** state, is notified to shutdown via the Unix signal `SIGINT`.
+On receipt of this signal, the ingester switches to `LEAVING`: in this state it cannot receive write requests anymore, while it could still receive read requests for series it has in memory.
+
```

Comment:
```suggestion
When a running ingester in the **`ACTIVE`** state is notified to shutdown via `SIGINT` or `SIGTERM` Unix signal, the ingester switches to `LEAVING` state. In this state it cannot receive write requests anymore, but it can still receive read requests for series it has in memory.
```